### DISCUSSION TITLE
Bagaimanakah cara meningkatkan Nilai Accuracy, Sensitivity, Specificity dan Precision Value dg Random Forest ?

### DISCUSSION THREAD

**Discussion by Micoalgo** [teamalgoritma/community#635] [Q&A] [answered]
Setelah dilakukan metode Random Forest sbb ;

<img width="328" alt="image" src="https://user-images.githubusercontent.com/123436963/231981438-419248e3-9328-4c08-96f3-d465dd3df49f.png">

<img width="402" alt="image" src="https://user-images.githubusercontent.com/123436963/231980094-baee2045-93c7-4355-ad97-47147595b2ea.png">

dihasilkan ;

<img width="418" alt="image" src="https://user-images.githubusercontent.com/123436963/231980324-638f6d16-0e2c-4876-bb54-98a43f3647fe.png">

dan setelah dilakukan prediksi thd data validation nya (data test dari data train nya) didapatkan hasill sbb;

<img width="366" alt="image" src="https://user-images.githubusercontent.com/123436963/231980601-c6c25983-ec70-40ff-9537-21823718589d.png">

<img width="372" alt="image" src="https://user-images.githubusercontent.com/123436963/231980528-9d4024f0-6276-43fc-9dca-76c9c8644d0a.png">

Bagaimanakah cara meningkatkan Nilai Accuracy, Sensitivity, Specificity dan Precision Value dg Random Forest tsb?
(Apakah harus ada perubahan pada parameter Fitting Model yaitu method, number , repeats nya atau yg lainnya ?)

<img width="249" alt="image" src="https://user-images.githubusercontent.com/123436963/231981204-a57d7eb9-890a-44aa-bc52-6be751bc441d.png">

Terimakasih

**Comment by IrfanChairurrachman** [comment]
Dear @Micoalgo ,

Untuk melakukan tuning model random forest menggunakan `caret` ada beberapa hal yang dapat dilakukan, antara lain:
1. Dengan mengubah nilai ntree (jumlah pohon yang dibentuk Random Forest) dengan cara memasukkan parameter ntree pada fungsi `train`, berikut referensi dari [askalago](https://askalgo.netlify.app/#tree-based-model)
2. Dapat juga dengan mengubah jumlah `mtry`, untuk referensi seperti di [askalago](https://askalgo.netlify.app/#tree-based-model)
3. Mencoba mengganti `parameter = 'rf'` di dalam fungsi `train` menggunakan model random forest dari library lain. Daftar model dapat dilihat di link [berikut](https://topepo.github.io/caret/available-models.html).  Semisal menggunakan `parameter = 'ranger'` atau jenis lainnya.

Sekian jawaban dari kami, jika masih terdapat kendala atau hal yang ingin ditanyakan dapat reply pesan berikut ini.

Best Regards,
Algoritma

**Reply by IrfanChairurrachman** [reply, marked as answer]
Dear @Micoalgo ,

Selain melakukan hyperparameter tuning, cara lain untuk meningkatkan performa model adalah memastikan tahapan preprocessing dan persiapan feature telah tepat. Semisal kita mempunyai feature tanggal, data tanggal kurang baik sebagai feature, sehingga kita dapat melakukan partisi/esktraksi informasi dari tanggal yang kita punya menjadi kolom/feature yang lebih berguna bagi model, seperti kolom nama bulan, kolom minggu, kolom hari ataupun kolom jam.

Contoh:
- `23-01-2023 14:44:00` -> 'Januari' (feature bulan) / 'Senin' (feature hari) / '14' (fitur jam) dsb (tergantung problem dari model)

Sekian jawaban dari kami, semoga membantu
Best Regards,
Team Algoritma

**Comment by Micoalgo** [comment]
Terimakasih jawabannya Mas Irfan
